### PR TITLE
Code cleanup for `Commands`

### PR DESCRIPTION
**Pull request by noplanman**
This PR is just for filling up the documentation and modifying some code style.
The commands in general need a bigger overhaul to prevent duplicate code and make it more sensible.

### CODE REVIEW

File: src/Commands/SlapCommand.php [hunk 1]
```diff
@@ -28,20 +66,22 @@ public function execute()
         $chat_id = $message->getChat()->getId();
         $message_id = $message->getMessageId();
         $text = $message->getText(true);
-        
-        $sender='@'.$message->getFrom()->getUsername();
 
-        $data = array();
-        $data['chat_id'] = $chat_id;
+        $sender = '@' . $message->getFrom()->getUsername();
```
Comment (MBoretto) [inline src/Commands/SlapCommand.php]: We should use the methods ->tryMention() from the class User.
https://github.com/akalongman/php-telegram-bot/blob/master/src/Entities/User.php#L63

File: src/Commands/SupergroupchatcreatedCommand.php [hunk 2]
```diff
@@ -1,27 +1,66 @@
...
+     * If this command is enabled
+     *
+     * @var boolean
+     */
     protected $enabled = true;
 
+    /**
+     * Execute command
+     *
+     * @todo $chat_id isn't defined!
```
Comment (MBoretto) [inline src/Commands/SupergroupchatcreatedCommand.php]: You are right this is an error..

File: src/Commands/GenericCommand.php [hunk 3]
```diff
@@ -1,41 +1,83 @@
...
+     * If this command is enabled
+     *
+     * @var boolean
+     */
     protected $enabled = true;
 
+    /**
+     * Execute command
+     *
+     * @todo This can't be right, as it always returns "Command: xyz not found.. :("
```
Comment (MBoretto) [inline src/Commands/GenericCommand.php]: This command is called with recursion here: 
https://github.com/akalongman/php-telegram-bot/blob/master/src/Telegram.php#L492
After that all the commands in the dirs has been parsed.

Comment (MBoretto) [inline src/Commands/GenericCommand.php]: Did you noticed that you can set if a command is public or not:
https://github.com/akalongman/php-telegram-bot/blob/master/src/Command.php#L24
This option is used to show or hide commads in  the /help 
https://github.com/akalongman/php-telegram-bot/blob/master/src/Commands/HelpCommand.php#L45

I think that this should be extended also to the execution. So public command can be executed by the user instead private one (As generic, Newchatparticipant, ... and so on) should be executed only by the library.

Comment (noplanman) [inline src/Commands/GenericCommand.php]: In my mind (as far as I've understood the different commands so far), there should be 3 different kinds of commands:
- Executed by the library / internal system
- Executed by the Admins
- Executed by the Users (& Admins obviously)

This way, we could separate these different commands more easily, as they have their own specific way of being executed. I see at the moment that many commands do pretty much the same thing, but each one uses lots of identical code, which really should be optimised.

Yes I saw that @ public vs. private. I think public should be the ones all Users can execute and private the ones only Admins can.

I haven't looked deeper into this, but maybe adding an extra layer could help out a lot:
```
-> Command
  |-> AdminCommand
    |-> SendToAllAdminCommand
  |-> UserCommand
    |-> WhoAmIUserCommand
  |-> SystemCommand
    |-> NewChatTitleSystemCommand
```

Something like that, you know what I mean?

**Comment by MBoretto** [comment]
Yes I think that this separation is needed.
With this schema we don't need anymore public variable.
System commands are not shown in the /help.
User commands  are visible to user in /help
Admin can see both user and admin commands in /help

**Comment by noplanman** [comment]
Yes. Also, I feel it's easier for developers to understand the different types of command.

Am I right that the System commands are given by Telegram?
Developers can't write their own, right?

**Comment by MBoretto** [comment]
They are connected to some chat action like `new_chat_participant` so they are given by telegram.
Anyway we should provide the possibility to overwrite the default one with their own. 
(this can be achieved with the custom folder commands).
For example genericcommand can be used to cought commands as param like /1 /green /the option chosen

**Comment by noplanman** [comment]
> possibility to overwrite the default one with their own

You mean when a System command like `new_chat_participant` is called, a developer can execute his own custom code? If so, absolutely!

Regarding the `GenericCommand`, that would be a `UserCommand`, right? So that any user can call `/whateveriwant` or `/maybeeventhis` and the developer can decide what happens to them. This would of course only be the case, if no separate command explicitly handles them.

Have I understood correctly?

**Pull request merged**